Data/SQLite/src/SessionImpl.cpp

```diff
@@ -205,16 +205,20 @@ void SessionImpl::open(const std::string& connect)
 
 void SessionImpl::close()
 {
-	if (_pDB) {
+	if (_pDB) 
+	{
 		int result = 0;
 		int times = 50;
-		do {
+		do 
+		{
 			result = sqlite3_close(_pDB);
-			if (result == SQLITE_BUSY) {
+			if (result == SQLITE_BUSY) 
```

Comment:
you can loop through all prepared statements using
```
sqlite3_next_stmt(_pDb, 0);
```
and then finalize each.
Note that there may be other `SQLITE_BUSY` on close reasons, like opened blob handles or backups. a proper solution would be to simulate such conditions and make sure they are handled correctly; preferably, in a separate pull.